core/trino-main/src/main/java/io/trino/execution/scheduler/ScaledWriterScheduler.java

```diff
@@ -133,6 +140,17 @@ private boolean isWriteThroughputSufficient()
         return writerInputBytes >= minWriterInputBytesToScaleUp;
     }
 
+    private boolean isMaxMemoryUtilizedAcrossAvailableTasks()
+    {
+        Collection<TaskStatus> writerTasks = writerTasksProvider.get();
+
+        writerTasks.stream().map(TaskStatus::getQueryMemoryReservation).forEach(System.out::println);
+
+        return writerTasks.stream()
+                .map(TaskStatus::getQueryMemoryReservation)
+                .allMatch(totalMemoryUsed -> totalMemoryUsed.toBytes() >= maxMemoryThresholdPerNode);
```

Comment:
Why is this `allMatch` instead of `anyMatch` ? Any one node's memory usage being high is enough to trigger the per-node memory usage exceeded exception